src/FSLibrary/MissionHistoryPubnetParallelCatchupV2.fs

```diff
@@ -0,0 +1,195 @@
+
+            if dict.TryGetValue(box "worker", &workerObj) then
+                match workerObj with
+                | :? Dictionary<obj, obj> as workerDict ->
+                    workerDict.[(box "stellar_core_image")] <- (box context.image)
+                    workerDict.[(box "replicas")] <- (box context.pubnetParallelCatchupNumWorkers)
+                    dict.[(box "worker")] <- box workerDict
+                | _ -> failwith "Unexpected YAML structure"
+            else
+                failwith "Unexpected YAML structure"
+
+            let mutable rangeObj = null
+
+            if dict.TryGetValue(box "range_generator", &rangeObj) then
+                match rangeObj with
+                | :? Dictionary<obj, obj> as rangeDict ->
+                    let mutable dictObj = null
+
+                    if rangeDict.TryGetValue(box "params", &dictObj) then
+                        match dictObj with
+                        | :? Dictionary<obj, obj> as paramsDict ->
+                            paramsDict.[(box "starting_ledger")] <- (box context.pubnetParallelCatchupStartingLedger)
+                            paramsDict.[(box "latest_ledger_num")] <- (box (GetLatestPubnetLedgerNumber()))
+                            rangeDict.[(box "params")] <- box paramsDict
+                        | _ -> failwith "Unexpected YAML structure"
+
+                    dict.[(box "range_generator")] <- box rangeDict
+                | _ -> failwith "Unexpected YAML structure"
+
+            dict
+        | _ ->
+            LogError "%s" (values.ToString())
+            failwith "Unexpected YAML structure"
+
+    // Serialize the updated values to a temporary file
+    let serializer =
+        SerializerBuilder()
+            .WithNamingConvention(CamelCaseNamingConvention.Instance)
+            .Build()
+
+    let tempFile = Path.GetTempFileName()
+    use writer = new StreamWriter(tempFile)
+    serializer.Serialize(writer, updatedValues)
+    writer.Flush()
+    // install the project, then delete the temporary helm values file
+    runCommand [| "helm"
+                  "install"
+                  helmReleaseName
+                  helmChartPath
+                  "--values"
+                  tempFile |]
+    |> ignore
+
+    File.Delete(tempFile)
+
+// Cleanup on exit
+let cleanup () =
+    if not cleanupCalled then
+        cleanupCalled <- true
+        LogInfo "Cleaning up resources..."
+
+        runCommand [| "helm"
+                      "uninstall"
+                      helmReleaseName |]
+        |> ignore
+
+System.AppDomain.CurrentDomain.ProcessExit.Add(fun _ -> cleanup ())
+
+Console.CancelKeyPress.Add
+    (fun _ ->
+        cleanup ()
+        Environment.Exit(0))
+
+let getJobMonitorStatus () =
+    try
+        use client = new HttpClient()
+
+        let response =
+            client
+                .GetStringAsync(
+                    "http://ssc-job-monitor.services.stellar-ops.com/status"
```

Comment:
Is this a stable DNS name? If so how? What if there are 2 SSC jobs running at the same time?

Reply:
In general we shouldn't have multiple SSC jobs running. If the need comes, I think it's easy to modify the DNS entry to add a wildcard matching to the CNAME record, or to route traffic to different job-monitor services via `paths` .